### DISCUSSION TITLE
QInput with QDate in a custom component

### DISCUSSION THREAD

**Discussion by Alen-Loforte** [quasarframework/quasar#15861] [General - Components / Directives / etc] [answered]
Hi guys
I'm having trouble creating a Custom component that has QInput with QDate. I tried following https://forum.quasar-framework.org/topic/6660/qinput-with-qdate-in-a-custom-component but with it didnt work. 
Here is my BaseComponent
[BaseDateInput.txt](https://github.com/quasarframework/quasar/files/11508710/BaseDateInput.txt)

When i click the component it opens Q-date without any problem, the problem happens when i try to choose a date. When i click a date it doesn't emit any event or updates the model-value.
Thanks for any help

**Comment by pdanpdan** [comment]
Put it in a codepen please

**Reply by Alen-Loforte** [reply]
Still trying to figure how codepen works

**Comment by pdanpdan** [comment]
You open any example in the docs with the codepen button.
You make your changes. You save

**Reply by Alen-Loforte** [reply]
Hi
I managed to replicate my issue on stackblitz here is the link:
https://stackblitz.com/edit/quasarframework-43bpuw?file=src/pages/BaseDateInput.vue

**Reply by pdanpdan** [reply]
you emit `@update:model-value="(date) => $emit('update:text', date)"` but use the component with `v-model` and it expects to receive `update:model-value`

**Reply by Alen-Loforte** [reply]
I tried putting model-value="date" instead of v-model="date", but when i use model-value="date" i cant select any date in the q-date component

**Comment by Alen-Loforte** [comment, marked as answer]
Ended up creating a function to emit the value to the parent and then created another function in the parent to receive the value.
